packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -473,7 +476,7 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
     const gasFeeData = await this.getBumpedGasFeeData()
     const maxGasPrice = this.parseGwei(this.maxGasPriceGwei)
     const priorityFeePerGasCap = this.parseGwei(this.priorityFeePerGasCap)
-    const isMaxReached = gasFeeData.gasPrice?.gt(maxGasPrice) || gasFeeData.maxPriorityFeePerGas?.gt(priorityFeePerGasCap)
+    const isMaxReached = this.gasPrice?.gte(maxGasPrice) || this.maxPriorityFeePerGas?.gte(priorityFeePerGasCap)
```

Comment:
This looks good. To be clear, was this changed because the old code had a bug, correct?

Reply:
old code:
return bumped gas fee data and check max. If it's below max, then then submit transaction and set `this.` properties. The value of getBumpedGasFeeData was not capped so this check was expected and not a bug

new code:
return bumped gas fee data with cap included. It's updated to check `this.` instead now because getBumpedGasFeeData will always returned the capped fee so it won't be able to get past the max fee check here if it's checking the returned value and therefore not be able to send tx with bumped gas fee. So it checks `this.` on the next time it has to boost to see if it already sent a tx with the max gas price

Reply:
See comment above about this change.